src/utils/time-entry-utils/timeEntryUtils.js

```diff
@@ -19,3 +22,13 @@ export const formatDate = (dateTime) => {
 export const formatDateTimeISO = (dateTime) => {
   return dayjs(dateTime).format('YYYY-MM-DDTHH:mm:ssZ');
 };
+
+export const removeTimecardContextEntry = (
+  timeEntries,
+  setTimeEntries,
+  removeAtIndex
+) => {
+  const newTimeEntries = deepClone(timeEntries);
```

Comment:
I really hope this deepClone is a proper deep copy and not one that doesn't copy functions and other non-serializable objects. If it is the latter it might be time to consider pulling in the use of a library like Lodash vs rolling our own functions and coming up against bugs in the future.

https://developer.mozilla.org/en-US/docs/Glossary/Deep_copy

Reply:
At the moment, `deepClone` is only use to copy objects or arrays, not functions. Do you think we need to rename the function to reflect that, or would you prefer us to use Lodash instead?

Reply:
Lodash actually looks pretty useful as it has lots of other helpful functions when needing to reconstruct data etc. So i'm all for using Lodash unless we're content with what we have now.